core/templates/dev/head/domain/exploration/ParamChangesObjectFactorySpec.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview unit tests for ParamChangesObjectFactory.
+ */
+
+import { TestBed } from '@angular/core/testing';
+
+import { ParamChange } from
+  'domain/exploration/ParamChangeObjectFactory';
+import { ParamChangesObjectFactory } from
+  'domain/exploration/ParamChangesObjectFactory';
+
+describe('ParamChanges Object Factory', () => {
+  let pcsof: ParamChangesObjectFactory;
+  const cArgs = {
+    parse_with_jinja: true,
+    value: ''
+  };
+  const gId = 'Copier';
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [ParamChangesObjectFactory]
+    });
+
+    pcsof = TestBed.get(ParamChangesObjectFactory);
+  });
+
+  it('should create a ParamChange array from a list of dictionaries',
+    () => {
+      let paramName = 'param_1';
+      let paramName2 = 'param_2';
+      let backendList = [{
+        customization_args: cArgs,
+        generator_id: gId,
+        name: paramName
+      },
+      {
+        customization_args: cArgs,
+        generator_id: gId,
+        name: paramName2
+      }];
+
+      let testOutcome: ParamChange[] = pcsof.createFromBackendList(backendList);
+
+      expect(testOutcome.length).toBe(2);
+      expect(testOutcome[0].customizationArgs).toEqual(cArgs);
```

Comment:
Hey @aks681,
Should there be functions in object factory for these? Like getLength(), getCustomizationArgs() etc.
Thanks

Reply:
Yeah, there should ideally, but a lot of the old exploration related object factories actually don't use that (eg: InteractionObjectFactory), and there are a lot of directives that use these, so changing it now would be more involved.